src/components/Coldcard/ColdcardSigner.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import moment from "moment";
+import { COLDCARD, ConfigAdapter } from "unchained-wallets";
+import { connect } from "react-redux";
+import { ColdcardSigningButtons } from ".";
+import { ColdcardPSBTReader } from "./ColdcardFileReader";
+import { downloadFile } from "../../utils";
+import { getWalletDetailsText } from "../../selectors/wallet";
+
+class ColdcardSigner extends Component {
+  handlePSBTDownloadClick = () => {
+    const { walletName, interaction, setActive } = this.props;
+    const body = interaction.request().toBase64();
+    const timestamp = moment().format("HHmm");
+    const filename = `${timestamp}-${walletName}.psbt`;
+    downloadFile(body, filename);
+    setActive();
+  };
+
+  handleWalletConfigDownloadClick = () => {
+    const { walletDetailsText } = this.props;
+    this.reshapeConfig(walletDetailsText);
+  };
+
+  // This tries to reshape it to a Coldcard Wallet Config via unchained-wallets
+  reshapeConfig = (walletDetails) => {
+    const walletConfig = JSON.parse(walletDetails);
+    const { startingAddressIndex } = walletConfig;
+    // If this is a config that's been rekeyed, note that in the name.
+    walletConfig.name =
```

Comment:
note to self/selves: this operation of "reshaping" feels like it would be a useful utility that's part of the config adapter/utility.